gis/gdal/Portfile

```diff
@@ -323,6 +324,13 @@ variant arrow description {Enable (Geo)Arrow IPC File Format / Stream and (Geo)P
                             -DOGR_ENABLE_DRIVER_ARROW=OFF   -DOGR_ENABLE_DRIVER_ARROW=ON \
                             -DOGR_ENABLE_DRIVER_PARQUET=OFF -DOGR_ENABLE_DRIVER_PARQUET=ON
     configure.args-append   -DARROW_USE_STATIC_LIBRARIES=OFF
+
+    # https://trac.macports.org/ticket/70298
+    if {[string match *gcc* ${configure.compiler}] \
+        && (${configure.build_arch} in [list i386 ppc])} {
+        configure.ldflags-append \
+                                -latomic
+    }
```

Comment:
This is not correct for universal builds. Use the facilities of the muniversal 1.0 portgroup which this port already includes. Optionally, upgrade to the muniversal 1.1 portgroup which uses different syntax but makes things simpler.

Reply:
I do not know its syntax, since universal builds are broken on `ppc` (because universal gcc is broken).

@Dave-Allured @nilason Does anyone have experience with universal builds?

P. S. Building gdal +universal with gcc would be quite a non-standard endeavor…

Reply:
What's needed is to add `-latomic` for any 32-bit architectures that are part of the build but not for any 64-bit architectures. One purpose of the muniversal portgroup is to provide a way to do that. Consult other ports using this portgroup for examples of how to use it, if the documentation in the comments at the top of the portgroup is unclear.

The muniversal 1.0 portgroup handles the universal build only and it is up to the Portfile to add corresponding code if the universal variant has not in fact been selected. The muniversal 1.1 portgroup simplifies it so that you only need to indicate what flags go with each arch and it adds them correctly regardless of whether or not the universal variant was selected.

Reply:
It does not actually do anything bad if we link with `libatomics` with a GCC build, it is required anyway due to 32-bit slice. It is not needed for 64-bit, but it won’t break it either.

Reply:
For years MacPorts always linked in libatomic for gcc builds regardless of architecture, if atomics were needed. 

Only recently was it felt necessary to only try to link in libatomic for 32 bit builds. This causes headaches and just opens the door for errors -- IMO just stop doing that, and link libatomic always for gcc builds if atomics are needed, like we always used to do.

libatomic tests for each required atomic action (during configure) and the library then uses compiler intrinsics if they exist, otherwise will fall back to handwritten atomics.

https://gitlab.com/gnutools/gcc/-/blob/trunk/libatomic/acinclude.m4

https://gitlab.com/gnutools/gcc/-/blob/trunk/libatomic/load_n.c#L29

Reply:
@kencu Ken, I am fine with using `libatomic` for all GCC builds. I thought it is cleaner to use it conditionally on the arch, but it is okay to use it like before. I will reopen the dedicated PR with this fix, dropping arch condition.